x-pack/platform/plugins/shared/inference/server/chat_complete/anonymization/regex_worker.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import Piscina from 'piscina';
+import { RegexAnonymizationTask } from '@kbn/inference-common';
+import { regexWorkerFilename } from './regex_worker_task';
+
+let worker: Piscina | undefined;
+
+export function initRegexWorker() {
+  worker = new Piscina({
+    filename: require.resolve('./regex_worker_wrapper.js'),
+    workerData: { fullpath: regexWorkerFilename },
+    maxThreads: 1,
```

Comment:
@dgieselaar any opinion on other constraints here like `resourceLimits`  or min/max threads.

Reply:
minThreads should be 0, i think maxThreads can be 2-3. will probably not use it but also doesn't hurt. Have you tested what happens when all threads are busy?